pkg/actuators/machine/actuator.go

```diff
@@ -461,6 +459,24 @@ func (a *Actuator) getMachineInstances(cluster *clusterv1.Cluster, machine *mach
 	return getRunningInstances(machine, client)
 }
 
+func (a *Actuator) getExistingMachineInstances(cluster *clusterv1.Cluster, machine *machinev1.Machine) ([]*ec2.Instance, error) {
```

Comment:
It's duplicating `getMachineInstances`. I would rather create two new variables `instanceFilterRunningInstances` and `instanceFilterExistingInstances` and pass the relevant filter through `getMachineInstances` and `getInstances` instead.